tensorflow_datasets/image/bccd.py

```diff
@@ -0,0 +1,128 @@
+"""
+BCCD (Blood Cell Count and Detection) Dataset is a small-scale dataset for blood cells detection.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow_datasets.public_api as tfds
+import tensorflow as tf
+import xml.etree.ElementTree as ET
+
+_CITATION = """
+@misc{shenggan_2019,
+    title={Shenggan/BCCD_Dataset},
+    url={https://github.com/Shenggan/BCCD_Dataset},
+    journal={GitHub},
+    author={Shenggan},
+    year={2019},
+    month={Nov}}
+"""
+
+_DESCRIPTION = """
+BCCD (Blood Cell Count and Detection) Dataset is a small-scale dataset for blood cells detection.
+12,500 augmented images of blood cells (JPEG) with accompanying cell type labels (CSV).
+There are approximately 3,000 images for each of 4 different cell types grouped into 4 different folders (according to cell type).
+The cell types are Eosinophil, Lymphocyte, Monocyte, and Neutrophil.
+"""
+
+_URL = "https://github.com/Shenggan/BCCD_Dataset"
+_URL_BCCD = "https://raw.githubusercontent.com/Shenggan/BCCD_Dataset/master/BCCD"
+_URL_TRAIN_TXT = _URL_BCCD + "/ImageSets/Main/train.txt"
+_URL_Val_TXT = _URL_BCCD + "/ImageSets/Main/val.txt"
+_URL_TEST_TXT = _URL_BCCD + "/ImageSets/Main/test.txt"
+_URL_IMG = _URL_BCCD + "/JPEGImages"
+_URL_XML = _URL_BCCD + "/Annotations"
+
+def parse_annotations(annotation_path):
+  root = ET.parse(annotation_path).getroot()
```

Comment:
All the file access has to go via tf.gfile 

        with tf.io.gfile.GFile(annotation_path, "rb") as f:
          root = ET.parse(f)